src/plugins/unified_search/public/query_string_input/query_string_input.tsx

```diff
@@ -179,23 +179,26 @@ export default class QueryStringInputUI extends PureComponent<QueryStringInputPr
   };
 
   private fetchIndexPatterns = debounce(async () => {
-    const stringPatterns = this.props.indexPatterns.filter(
-      (indexPattern) => typeof indexPattern === 'string'
-    ) as string[];
-    const objectPatterns = this.props.indexPatterns.filter(
-      (indexPattern) => typeof indexPattern !== 'string'
-    ) as DataView[];
+    const [objectPatterns = [], stringPatterns = []] = partition<
+      QueryStringInputProps['indexPatterns'][number],
+      DataView
+    >(this.props.indexPatterns || [], (indexPattern): indexPattern is DataView => {
+      return indexPattern instanceof DataView;
```

Comment:
In Graph tests this condition is `false` and a mocked data view falls into `stringPatterns` bucket although it should be in `objectPatterns`